backend/src/Entity/DocumentCategory.php

```diff
@@ -37,4 +52,42 @@ public function setName(string $name): self
 
         return $this;
     }
+
+    public function getParent(): ?self
+    {
+        return $this->parent;
+    }
+
+    public function setParent(?self $parent): self
+    {
+        $this->parent = $parent;
+
+        return $this;
+    }
+
+    public function getSubcategories(): Collection
+    {
+        return $this->subcategories;
+    }
+
+    public function addSubcategory(self $subcategory): self
+    {
+        if (!$this->subcategories->contains($subcategory)) {
+            $this->subcategories->add($subcategory);
+            $subcategory->setParent($this);
```

Comment:
you should include a check that avoids cyclical relationships (eg assigning one of the ancestors as a child